rust/agent-anthropic/src/lib.rs

```diff
@@ -0,0 +1,98 @@
+use std::{env, sync::Arc};
+
+use anthropic::{
+    client::Client,
+    config::AnthropicConfig,
+    types::CompleteRequestBuilder,
+    AI_PROMPT, HUMAN_PROMPT,
+};
+
+
+use agent::{
+    common::{
+        async_trait::async_trait,
+        eyre::{bail, Result},
+    },
+    Agent, AgentIO, GenerateOptions,
+};
+
+/// An agent running on OpenAI
+///
+/// The environment variable OPENAI_API_KEY must be set to use these agents.
+pub struct AnthropicAgent {
+    /// The OpenAI name for a model including any tag e.g. "llama2:13b"
+    ///
+    /// Used as the required `model` parameter in each request to `POST /api/generate`
+    /// (along with `prompt`).
+    model: String,
+
+}
+
+impl AnthropicAgent {
+    /// Create a OpenAI-based agent
+    pub fn new(model: String) -> Self {
+        Self {
+            model,
+        }
+    }
+   fn supported_inputs(&self) -> &[AgentIO] {
+        &[AgentIO::Text]
+    }
+
+    fn supported_outputs(&self) -> &[AgentIO] {
+        &[AgentIO::Text]
+    }
+}
+
+#[async_trait]
+impl Agent for AnthropicAgent {
+
+    fn provider(&self) -> String {
+        "anthropic".to_string()
+    }
+
+    fn model(&self) -> String {
+        self.model.clone()
+    }
+
+    // fn supported_inputs(&self) -> &[AgentIO] {
+    //     &[AgentIO::Text]
+    // }
+
+    // fn supported_outputs(&self) -> &[AgentIO] {
+    //     &[AgentIO::Text]
+    // }
+
+    async fn text_to_text(&self, instruction: &str, options: &GenerateOptions) -> Result<String> {
```

Comment:
At the top of this function you should render the prompt template with the instruction. This is how we pass context variables into the custom prompts. See `agent-openai` crate.

```rust
        let (system_prompt, user_prompt) = self.render_prompt(
            &options.prompt_name,
            json!({
                "user_instruction": instruction
            }),
        )?;
```